src/plugins/foxPage/hooks/useFarmingApr.ts

```diff
@@ -81,5 +135,60 @@ export const useFarmingApr = () => {
     })()
   }, [blockNumber, uniV2LPContract, farmingRewardsContractV2, farmingRewardsContractV4])
 
-  return { isFarmingAprV2Loaded, isFarmingAprV4Loaded, farmingAprV2, farmingAprV4 }
+  const addLiquidity = useCallback(
+    async (foxAmount: string, ethAmount: string) => {
+      try {
+        if (!connectWalletEthAddress || !uniswapRouterContract) return
+        const data = uniswapRouterContract.interface.encodeFunctionData('addLiquidityETH', [
+          FOX_TOKEN_CONTRACT_ADDRESS,
+          toBaseUnit(foxAmount, 18),
+          calculateSlippageMargin(slippagePercentage, foxAmount),
+          calculateSlippageMargin(slippagePercentage, ethAmount),
+          connectWalletEthAddress,
+          Date.now() + 1200000,
+        ])
+
+        const tx = {
+          from: connectWalletEthAddress,
+          to: UNISWAP_V2_ROUTER_ADDRESS,
+          data,
+          value: toHex(toBaseUnit(ethAmount, 18)),
+        }
+
+        const { gasLimit, gasPrice } = await getBufferedGas(tx)
+        if (gasLimit && gasPrice) {
+          // dispatch({
+          //   type: LpActions.SET_TX_FEE,
+          //   payload: bn(gasLimit).times(gasPrice).toFixed()
+          // })
+          const ethBalance = await ethersProvider.getBalance(connectWalletEthAddress)
+          if (
+            bnOrZero(ethBalance.toString())
+              .minus(bnOrZero(toBaseUnit(ethAmount as string, 18)))
+              .lt(bnOrZero(gasLimit).times(gasPrice).toFixed())
+          ) {
+            throw new Error('Not enough ETH for gas')
+          }
+
+          const nonce = await ethersProvider.getSigner().getTransactionCount()
+          const lpTx = await ethersProvider.getSigner().sendTransaction({
+            from: connectWalletEthAddress,
+            to: tx.to,
+            data: tx.data,
+            value: tx.value,
+            gasLimit: toHex(gasLimit),
+            gasPrice: toHex(gasPrice),
+            nonce,
+            chainId: 1,
```

Comment:
I get it we know what we're doing here, controlled context etc, but pls no hardcoded `chainId`, you can either:

- Decode the `chainReference` part if using a CAIP-vernacular `AccountId`
- Preferred, use the new `useEVM` hook to get the CAIP chainReference part (a.k.a ETH network a.k.a EVM chain ID). That might however produce unwanted behavior if you're connected on any ETH network other than mainnet

Reply:
Can't we use `signAndBroadcastTransaction` here rather than relying on ethersjs? We do need to reimplement it to call view methods on contracts, but calling such methods should be done through chain-adapters if possible.
You'll also leverage the behavior of automagically switching the wallet to the right chainId for the connected chain-adapter.